### PR TITLE
Improve terraform deployment docs

### PR DESCRIPTION
**Pull request by nbeloglazov**

### CODE REVIEW

File: docs/it-manual/sre-playbook/terraform-deploy-system.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+Terraform - is an infrastructure as code tool that lets you define both cloud and on-prem resources in human-readable configuration files that you can version, reuse, and share. Civiform provides Terraform configuration files that allow to deploy Civiform on Azure and AWS. Knowledge of Terraform is not required to run them but here is the high-level [Terraform overview](https://www.terraform.io/intro) nevertheless. 
+
 ## Setup
 
 ### Outside configuration
-You will need couple of values that are configured outside of Civiform before you start the setup.
-- [ ] Admin auth client_id, client_secret, and discovery_uri. See [setting up Azure AD for an example](#setting-up-azure-ad)
-- [ ] Applicant auth client_id, client_secret, and discovery_uri. See setting up the [Authentication Providers](../../contributor-guide/developer-guide/authentication-providers.md)
-- [ ] Domain name for your deployment
+You will need couple of values that are configured outside of Civiform before you start the setup. Some of the steps are optional meaning that you can bring up staging environment and get app working, but they need to be completed for production setup.
```
Comment (azizova-leyla) [inline docs/it-manual/sre-playbook/terraform-deploy-system.md:8]: I'm pretty sure the app will not start if these values are empty. discovery_uri need to be a valid url.

File: docs/it-manual/sre-playbook/terraform-deploy-system.md [hunk 2]
```diff
@@ -26,7 +32,11 @@ You will need couple of values that are configured outside of Civiform before yo
 
 ## Troubleshooting
 
-#### Terraform fails
+#### Terraform fails with error "Provider produced inconsistent final plan"
+
+This error can happen when running `bin/setup` for the first time. If you set it - re-run `bin/setup`. This is a known Terraform [bug](https://github.com/hashicorp/terraform-provider-aws/issues/19583).
```
Comment (azizova-leyla) [inline docs/it-manual/sre-playbook/terraform-deploy-system.md:37]: typo: if you **see** it.

File: docs/it-manual/sre-playbook/terraform-deploy-system.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
 # Terraform deploy system
 
+Terraform - is an infrastructure as code tool that lets you define both cloud and on-prem resources in human-readable configuration files that you can version, reuse, and share. Civiform provides Terraform configuration files that allow to deploy Civiform on Azure and AWS. Knowledge of Terraform is not required to run them but here is the high-level [Terraform overview](https://www.terraform.io/intro) nevertheless.
```
Comment (azizova-leyla) [inline docs/it-manual/sre-playbook/terraform-deploy-system.md:3]: no need for dash here. "Terrafor is an infrastucture as code..."

**Review by azizova-leyla — CHANGES_REQUESTED** [review]

File: docs/it-manual/sre-playbook/terraform-deploy-system.md [hunk 4]
```diff
@@ -16,7 +18,11 @@ You will need couple of values that are configured outside of Civiform before yo
 3. Find the version that you want to deploy on [Github](https://github.com/civiform/civiform/releases)
 4. Copy the civiform\_config.example.sh into civiform\_config.sh and fill out the missing values. You can get a sense of required values depending on your cloud provider by looking at [staging-azure](https://github.com/civiform/staging-azure-deploy/blob/main/civiform_config.sh) or [staging-aws](https://github.com/civiform/staging-aws-deploy/blob/main/civiform_config.sh) configs.
 5. Run the `bin/doctor --tag=<version>` and install the dependencies
-6. Run `bin/setup --tag=<version>`
+6. Run `bin/setup --tag=<version>`. What to expect:
+    * Runs 5-10 minutes.
+    * Terraform brings up resources in cloud (database, network, server, etc).
+    * Asks confirmation few times before creating resources listing everything that will be created. 
+    * Safe to re-run script if it fails.
```
Comment (avrittrohwer) [inline docs/it-manual/sre-playbook/terraform-deploy-system.md:25]: Maybe worth adding that until https://github.com/cn-terraform/terraform-aws-logs-s3-bucket/issues/6 is resolved, setup will have to be run twice

**Review by avrittrohwer — APPROVED** [review]
LGTM, could you mark this as fixing https://github.com/civiform/civiform/issues/3634 as well?

File: docs/it-manual/sre-playbook/terraform-deploy-system.md (hunk 2 above)
Comment (nbeloglazov) [reply docs/it-manual/sre-playbook/terraform-deploy-system.md]: Done.

File: docs/it-manual/sre-playbook/terraform-deploy-system.md (hunk 4 above)
Comment (nbeloglazov) [reply docs/it-manual/sre-playbook/terraform-deploy-system.md]: Done.

File: docs/it-manual/sre-playbook/terraform-deploy-system.md (hunk 1 above)
Comment (nbeloglazov) [reply docs/it-manual/sre-playbook/terraform-deploy-system.md:8]: Well. We have default dummy values in `civiform_config.example.sh`. You don't have to change them with your own real discovery uri so server can still start. Maybe instead we should have default config with no auth at all (have it disabled). And enabling auth would be a separate step, after you got staging running.

**Pull request merged**